server/src/main/java/org/jboss/as/server/ServerEnvironment.java

```diff
@@ -332,154 +335,183 @@ public ServerEnvironment(final String hostControllerName, final Properties props
         // Java system-wide extension dirs
         javaExtDirs = getFilesFromProperty(JAVA_EXT_DIRS, props);
 
-        // Must have HOME_DIR
-        homeDir = getFileFromProperty(HOME_DIR, props);
-        if (homeDir == null) {
-            throw ServerLogger.ROOT_LOGGER.missingHomeDirConfiguration(HOME_DIR);
-        }
-        if (!homeDir.exists() || !homeDir.isDirectory()) {
-            throw ServerLogger.ROOT_LOGGER.homeDirectoryDoesNotExist(homeDir);
-        }
-
-        @SuppressWarnings("deprecation")
-        File tmp = getFileFromProperty(MODULES_DIR, props);
-        if (tmp == null) {
-            tmp = new File(homeDir, "modules");
-        } else if (!tmp.exists() || !tmp.isDirectory()) {
-            throw ServerLogger.ROOT_LOGGER.modulesDirectoryDoesNotExist(tmp);
-        }
-        modulesDir = tmp;
+        if ( launchType.equals( LaunchType.SELF_CONTAINED ) ) {
+            homeDir = new File(WildFlySecurityManager.getPropertyPrivileged("user.dir", "."));
+            serverBaseDir = new File(WildFlySecurityManager.getPropertyPrivileged("user.dir", "."));
+            serverLogDir = new File(WildFlySecurityManager.getPropertyPrivileged("user.dir", "."));
 
-        configureBundlesDir(props.getProperty(BUNDLES_DIR), props);
 
-        tmp = getFileFromProperty(SERVER_BASE_DIR, props);
-        if (tmp == null) {
-            tmp = new File(homeDir, standalone ? "standalone" : "domain/servers/" + serverName);
-        }
-        if (standalone) {
-            if (!tmp.exists()) {
-                throw ServerLogger.ROOT_LOGGER.serverBaseDirectoryDoesNotExist(tmp);
-            } else if (!tmp.isDirectory()) {
-                throw ServerLogger.ROOT_LOGGER.serverBaseDirectoryIsNotADirectory(tmp);
+            try {
+                File tmpDir = File.createTempFile( "wildfly-self-contained", ".d" );
+                if ( tmpDir.exists() ) {
+                    if ( ! tmpDir.delete() ) {
```

Comment:
I just realized this might not work. It will error if the directory is not empty. We'd likely need a recursive delete.